sdk/core/azure-core/azure/core/exceptions.py

```diff
@@ -333,6 +333,16 @@ def _parse_odata_body(error_format, response):
             pass
         return None
 
+    def __str__(self):
+        retval = super(HttpResponseError, self).__str__()
+        try:
+            body = self.response.text()
+            if body and not self.error:
+                return "{}. Error: {}.".format(retval, body)[:2048]
```

Comment:
I would not call the body "Error" - it would be something more like "data" or "body". Or "details".

Reply:
What happens for an error if you specified "stream=True"?

Reply:
Sounds good! Actually do you think it's a good idea to go with "content", since that's what we're moving our "body" attr on the response to be?

Reply:
> What happens for an error if you specified "stream=True"?

Good question. I asked this question to myself too. 

But I see in our existing code, we already have 

odata_json = json.loads(response.text())

I don't have concern for this. :)

Reply:
there's a [separate issue](https://github.com/Azure/azure-sdk-for-python/issues/18086) for streamed error responses, where basically there's no body. I had a pr out for this at one point, but still haven't merged it. This suffers from the same issue as the rest of the response, but we won't throw any errors, we just won't have the body information in the message